crates/sui-indexer/src/handlers/pruner.rs

```diff
@@ -51,6 +52,25 @@ impl Pruner {
                 (min_epoch, max_epoch) = self.store.get_available_epoch_range().await?;
             }
 
+            // Update watermark lower bounds before doing actual pruning
+            let new_epoch_bound = max_epoch.saturating_sub(self.epochs_to_keep - 1);
+            let (cp, tx) = self
+                .store
+                .get_min_cp_and_tx_for_epoch(new_epoch_bound)
+                .await?;
+            let timestamp_ms = chrono::Utc::now().timestamp_millis();
```

Comment:
Rather than have the pruning or indexing service decide the timestamp, we should probably use the database's notion of time, otherwise you can get into trouble because of clock drift between the indexer/pruner/GraphQL.

Reply:
Also, I'm not seeing the logic to actually heed this timestamp or configure the delay for it. I expected it somewhere near here, but maybe I'm missing something?